### PR TITLE
Add ability to run apiDiff for incompatible changes

### PR DESCRIPTION
**Pull request by jimmyjames**
### Changes

Adds the `apiDiff` task now available through the [OSS Plugin](https://github.com/auth0/oss-library-gradle-plugin/blob/master/src/main/groovy/com/auth0/gradle/oss/LibraryPlugin.groovy#L216)

### References

- https://github.com/auth0/oss-library-gradle-plugin/pull/25

### Checklist

- [X] I have read the [Auth0 general contribution guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md)
- [X] I have read the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md)
- [X] All existing and new tests complete without errors

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -22,7 +21,49 @@ jobs:
...
+      # Download and cache dependencies
+      - restore_cache:
+          keys:
+          - v1-dependencies-{{ checksum "build.gradle" }}
+          # fallback to using the latest cache if no exact match is found
+          - v1-dependencies-
+      # run apiDiff task
+      - run: ./gradlew clean apiDiff
+      - store_artifacts:
+         path: build/reports/apiDiff/apiDiff.txt
```
Comment (lbalmaceda) [inline .circleci/config.yml:37]: I thought the output was going to be HTML. Can we upload that one as well? as it's easier to browse through that report vs plain text

File: .circleci/config.yml [hunk 2]
```diff
@@ -22,7 +21,49 @@ jobs:
...
+  run-api-diff:
+    steps:
+      - checkout
+      - run: chmod +x gradlew
+      # Download and cache dependencies
+      - restore_cache:
+          keys:
+          - v1-dependencies-{{ checksum "build.gradle" }}
+          # fallback to using the latest cache if no exact match is found
+          - v1-dependencies-
```
Comment (lbalmaceda) [inline .circleci/config.yml:33]: all these can be part of a single reusable step. Then build and test would run `clean check` & upload coverage, and apiDiff would run `clean apiDiff`. It should simplify a bit the script.

**Review by lbalmaceda — CHANGES_REQUESTED** [review]

**Review by lbalmaceda — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (jimmyjames) [reply .circleci/config.yml:31]: Uploading both html and txt reports now.

File: .circleci/config.yml (hunk 2 above)
Comment (jimmyjames) [reply .circleci/config.yml]: Updated to extract common steps.

**Pull request merged**